index.js

```diff
@@ -358,9 +359,7 @@ Buffer.isBuffer = function isBuffer (b) {
 }
 
 Buffer.compare = function compare (a, b) {
-  if (isInstance(a, Uint8Array)) a = Buffer.from(a, a.offset, a.byteLength)
-  if (isInstance(b, Uint8Array)) b = Buffer.from(b, b.offset, b.byteLength)
```

Comment:
Note that these lines were reallocating the entire buffer __every__ time, regardless of whether the inputs were Uint8Arrays or Buffers. This caused a massive slowdown when comparing large buffers.